libs/luci-lib-nixio/src/tls-context.c

```diff
@@ -54,20 +54,20 @@ static int nixio_tls_ctx(lua_State * L) {
 	lua_setmetatable(L, -2);
 
 	if (!strcmp(method, "client")) {
-		*ctx = SSL_CTX_new(TLSv1_client_method());
+		*ctx = SSL_CTX_new(TLS_client_method());
 	} else if (!strcmp(method, "server")) {
-		*ctx = SSL_CTX_new(TLSv1_server_method());
+		*ctx = SSL_CTX_new(TLS_server_method());
 	} else {
 		return luaL_argerror(L, 1, "supported values: client, server");
 	}
 
 	if (!(*ctx)) {
 		return luaL_error(L, "unable to create TLS context");
 	}
-
-#ifdef WITH_CYASSL
+	/* Setting the minimum version of SSL */
+	SSL_CTX_set_min_proto_version(*ctx,TLS1_2_VERSION);
+	/* Setting do not verify certificate */
 	SSL_CTX_set_verify(*ctx, SSL_VERIFY_NONE, NULL);
```

Comment:
Here's one point we should be looking to change.  There's not much of a point in using a TLS connection if we are not verifying the server certificate.